data/helpers.d/systemd

```diff
@@ -14,103 +14,106 @@
 #
-# usage: ynh_systemd_action [-n service_name] [-a action] [ [-l "line to match"] [-p log_path] [-t timeout] [-e length] ]
+# usage: ynh_systemd_action [--service_name=service_name] [--action=action] [ [--line_match="line to match"] [--log_path=log_path] [--timeout=300] [--length=20] ]
 # | arg: -n, --service_name= - Name of the service to start. Default : $app
 # | arg: -a, --action=       - Action to perform with systemctl. Default: start
 # | arg: -l, --line_match=   - Line to match - The line to find in the log to attest the service have finished to boot. If not defined it don't wait until the service is completely started. WARNING: When using --line_match, you should always add `ynh_clean_check_starting` into your `ynh_clean_setup` at the beginning of the script. Otherwise, tail will not stop in case of failure of the script. The script will then hang forever.
 # | arg: -p, --log_path=     - Log file - Path to the log file. Default : /var/log/$app/$app.log
 # | arg: -t, --timeout=      - Timeout - The maximum time to wait before ending the watching. Default : 300 seconds.
 # | arg: -e, --length=       - Length of the error log : Default : 20
+#
+# Requires YunoHost version 3.5.0 or higher.
 ynh_systemd_action() {
     # Declare an array to define the options of this helper.
     local legacy_args=nalpte
-    declare -Ar args_array=( [n]=service_name= [a]=action= [l]=line_match= [p]=log_path= [t]=timeout= [e]=length= )
+    local -A args_array=( [n]=service_name= [a]=action= [l]=line_match= [p]=log_path= [t]=timeout= [e]=length= )
     local service_name
     local action
     local line_match
     local length
     local log_path
     local timeout
-
     # Manage arguments with getopts
     ynh_handle_getopts_args "$@"
-
-    local service_name="${service_name:-$app}"
-    local action=${action:-start}
-    local log_path="${log_path:-/var/log/$service_name/$service_name.log}"
-    local length=${length:-20}
-    local timeout=${timeout:-300}
+    service_name="${service_name:-$app}"
+    action=${action:-start}
+    line_match=${line_match:-}
+    length=${length:-20}
+    log_path="${log_path:-/var/log/$service_name/$service_name.log}"
+    timeout=${timeout:-300}
 
     # Start to read the log
-    if [[ -n "${line_match:-}" ]]
+    if [[ -n "$line_match" ]]
     then
         local templog="$(mktemp)"
         # Following the starting of the app in its log
-        if [ "$log_path" == "systemd" ] ; then
+        if [ "$log_path" == "systemd" ]
+        then
             # Read the systemd journal
             journalctl --unit=$service_name --follow --since=-0 --quiet > "$templog" &
             # Get the PID of the journalctl command
             local pid_tail=$!
         else
             # Read the specified log file
-            tail -F -n0 "$log_path" > "$templog" 2>&1 &
+            tail --follow --retry --lines=0 "$log_path" > "$templog" 2>&1 &
```

Comment:
```suggestion
            tail --follow=name --retry --lines=0 "$log_path" > "$templog" 2>&1 &
```

From man:
> **-F**     same as **--follow**=_name_ **--retry**

> **-f**, **--follow**[={name|descriptor}]
                    output appended data as the file grows;
                    an absent option argument means 'descriptor'

NB: without _name_ we got a warning:
> tail: warning: --retry only effective for the initial open

Reply:
Oh ok, if you tested it then.
`2>&1` That's probably why I missed the error...